subsys/emds/emds.c

```diff
@@ -253,7 +288,23 @@ int emds_prepare(void)
 	(void)emds_entries_size(&size);
 
 	rc = emds_flash_prepare(&emds_flash, size);
-	if (rc) {
+
+	/** If flash is empty after prepare, imediatley make an
+	 * extra copy of the current entries as backup before the
+	 * next EMDS event in case it should fail
+	 */
+	if (IS_ENABLED(CONFIG_EMDS_AUTO_RECOVER) && rc == 1) {
+
+		ssize_t free_flash = emds_flash_free_space_get(&emds_flash);
+
+		/* Must be enough room for shutdown storage */
+		if (free_flash >= size * 2) {
```

Comment:
What happens if we don't have enough flash? Will we still have a backup?

Reply:
No, I wrote in the docs that it is recommended to ensure that the total flash size should be equal or larger than twice the size that is stored on a single shutdown. This is something that a user should really do anyways since not doing this will make the flash erase on every boot. If we do a backup and the entries stored is more than half the size of the entire flash, emds will fail to store on the actual shutdown do to lacking space.

Reply:
Sure, I agree 👍